sdk/examples/intkey_jvm_sc/sawtooth_intkey/cli/load.py

```diff
@@ -41,31 +68,26 @@ def _split_batch_list(batch_list):
 
 def do_load(args):
     print("Do load")
-    with open(args.filename, 'rb') as fd:
+    with open(args.filename, mode='rb') as fd:
         batches = batch_pb2.BatchList()
         batches.ParseFromString(fd.read())
 
-    stream = Stream(args.url)
-    futures = []
     start = time.time()
-
+    futures = []
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=5)
     for batch_list in _split_batch_list(batches):
-        future = stream.send(
-            message_type=Message.CLIENT_BATCH_SUBMIT_REQUEST,
-            content=batch_list.SerializeToString())
-        futures.append(future)
+        fut = executor.submit(post_batches, args.url, batch_list)
+        futures.append(fut)
 
     for future in futures:
```

Comment:
I wonder if we can do something better than the prior implementation by using concurrent.futures.as_completed. If it was ```for future in as_completed(futures):``` if there was wide variation in the number of txns per batch or the complexity of those txns, the total time it would take to load would be less.